prov/rxm/src/rxm_ep.c

```diff
@@ -628,8 +628,7 @@ static int rxm_ep_discard_recv(struct rxm_ep *rxm_ep, struct rxm_rx_buf *rx_buf,
 	RXM_DBG_ADDR_TAG(FI_LOG_EP_DATA, "Discarding message",
 			 rx_buf->unexp_msg.addr, rx_buf->unexp_msg.tag);
 
-	dlist_insert_tail(&rx_buf->repost_entry,
-			  &rx_buf->ep->repost_ready_list);
+	rxm_rx_buf_release(rx_buf->ep, rx_buf, 1);
```

Comment:
Similar comment as above.  We're replacing a direct call with one that inserts a branch into the code path.

Reply:
@shefty Thanks for clarification! I didn't thought about it from the performance side. From this point direct calls to those functions makes sense to me.